pkg/crinit/aggregated/aggregated.go

```diff
@@ -0,0 +1,399 @@
+		return err
+	}
+
+	glog.V(4).Info("Successfully created cluster role")
+
+	// Create a Kubernetes cluster role binding from the default service account
+	// in your namespace to the cluster role you just created
+	glog.V(4).Infof("Creating cluster role binding %v", clusterRole)
+
+	crb1, crb2, err, err2 := createClusterRoleBindings(cmdOut, clientset, opts)
+
+	if err != nil {
+		glog.V(4).Infof("Failed to create cluster role binding %v: %v", crb1, err)
+		return err
+	} else if err2 != nil {
+		glog.V(4).Infof("Failed to create cluster role binding %v: %v", crb2, err2)
+		return err
+	}
+
+	fmt.Fprintln(cmdOut, " done")
+	glog.V(4).Info("Successfully created cluster role binding")
+	return nil
+}
+
+// createServiceAccount handles the creation of the service account for the
+// the cluster registry to be used with RBAC.
+func createServiceAccount(cmdOut io.Writer, clientset client.Interface,
+	opts *aggregatedClusterRegistryOptions) (*v1.ServiceAccount, error) {
+
+	sa := &v1.ServiceAccount{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      serviceAccount,
+			Namespace: opts.ClusterRegistryNamespace,
+			Labels:    util.ComponentLabel,
+		},
+	}
+
+	if opts.DryRun {
+		return sa, nil
+	}
+
+	return clientset.CoreV1().ServiceAccounts(opts.ClusterRegistryNamespace).Create(sa)
+}
+
+// createClusterRole creates the cluster role for the operations we will allow
+// on our resources
+func createClusterRole(cmdOut io.Writer, clientset client.Interface,
+	opts *aggregatedClusterRegistryOptions) (*rbacv1.ClusterRole, error) {
+
+	rule := rbacv1.PolicyRule{
+		Verbs:     clusterRoleVerbs,
+		APIGroups: clusterRoleAPIGroup,
+		Resources: clusterRoleResources,
+	}
+
+	cr := &rbacv1.ClusterRole{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:   clusterRole,
+			Labels: util.ComponentLabel,
+		},
+		Rules: []rbacv1.PolicyRule{rule},
+	}
+
+	if opts.DryRun {
+		return cr, nil
+	}
+
+	return clientset.RbacV1().ClusterRoles().Create(cr)
+}
+
+// createClusterRoleBinding creates the cluster role binding for the operations we will allow
+// on our resources
+func createClusterRoleBindings(cmdOut io.Writer, clientset client.Interface,
+	opts *aggregatedClusterRegistryOptions) (*rbacv1.ClusterRoleBinding,
+	*rbacv1.ClusterRoleBinding, error, error) {
+
+	// cluster role binding for clusterregistry.k8s.io:apiserver cluster role
+	crb1 := &rbacv1.ClusterRoleBinding{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:   clusterRoleBindingAPIServer,
+			Labels: util.ComponentLabel,
+		},
+		Subjects: []rbacv1.Subject{
+			rbacv1.Subject{
+				Kind:      "ServiceAccount",
+				APIGroup:  "",
+				Name:      serviceAccount,
+				Namespace: opts.ClusterRegistryNamespace,
+			},
+		},
+		RoleRef: rbacv1.RoleRef{
+			APIGroup: "rbac.authorization.k8s.io",
+			Kind:     "ClusterRole",
+			Name:     clusterRole,
+		},
+	}
+
+	// cluster role binding for system:auth-delegator cluster role
+	crb2 := &rbacv1.ClusterRoleBinding{
```

Comment:
Maybe authDelegatorCRB (or authDelegatorRoleBinding)?

And call the previous one apiServerCRB?